src/main/java/com/cloudbees/jenkins/github/AccessTokenCredential.java

```diff
@@ -0,0 +1,109 @@
+package com.cloudbees.jenkins.github;
+
+import com.cloudbees.plugins.credentials.CredentialsDescriptor;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AutoCompletionCandidates;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.kohsuke.github.GHAuthorization;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class AccessTokenCredential extends BaseStandardCredentials implements StandardUsernameCredentials {
+
+    private final String githubServer;
+
+    private final String username;
+
+    private final Secret token;
```

Comment:
@stphenc my intent here was to offer option to create a token from jenkins UI, using a classic username/password authentication to github token generation API. The token then for sure can be considered being some username/password item, but the related UI is a distinct one.